modules/apps/transfer/types/trace.go

```diff
@@ -70,6 +74,24 @@ func (dt DenomTrace) GetFullDenomPath() string {
 	return dt.GetPrefix() + dt.BaseDenom
 }
 
+// extractPathAndBaseFromFullDenom returns the trace path and the base denom from
+// the elements that constitute the complete denom.
+func extractPathAndBaseFromFullDenom(fullDenomItems []string) (string, string) {
+	var path []string
+	var baseDenom []string
+	length := len(fullDenomItems)
+	for i := 0; i < length; i = i + 2 {
+		if i < length-1 && length > 2 && fullDenomItems[i] == PortID {
+			path = append(path, fullDenomItems[i], fullDenomItems[i+1])
+		} else {
```

Comment:
Nice work @crodriguezvega! It should be noted this introduces undocumented changes. Specifically the usage of `PortID`. ICS20 does not specify the PortID that should be used for the transfer channel. See [specification callbacks](https://github.com/cosmos/ibc/tree/master/spec/app/ics-020-fungible-token-transfer#channel-lifecycle-management). This is also true in our [code](https://github.com/cosmos/ibc-go/blob/main/modules/apps/transfer/ibc_module.go#L57). This code now assumes the PortID to be `transfer`. This should be enforced (in channel handshakes) if this the intended behaviour

I'm unaware of any chains which have changed the portID, but last I checked, Agoric used a modified portID in testing

Reply:
Nice catch @colin-axner ! Unfortunately without a standardized port, it's impossible to correctly differentiate between what is trace info and what is the base denom. Since the trace is largely internal information for a chain and will not affect correctness, I think we can live with the fact that non-standard ports and their subsequent traces will be part of the base denom.

Reiterating for future viewers that this will not affect correctness of the protocol. But when querying trace information, any nonstandard trace will be in the base denom